main/src/library/Iterable.flix

```diff
@@ -25,3 +25,43 @@ pub lawless class Iterable[t: Type -> Type] {
     pub def iterator(t: t[a]): Iterator[a] & Impure
 
 }
+
+instance Iterable[Array] {
+    pub def iterator(t: Array[a]): Iterator[a] & Impure =
+        Array.toIterator(t)
+}
+
+instance Iterable[List] {
+    pub def iterator(t: List[a]): Iterator[a] & Impure =
+        List.toIterator(t)
+}
+
+instance Iterable[LazyList] {
+    pub def iterator(t: LazyList[a]): Iterator[a] & Impure =
+        LazyList.toIterator(t)
+}
+
+instance Iterable[MutList] {
+    pub def iterator(t: MutList[a]): Iterator[a] & Impure =
+        t |> MutList.toList |> List.toIterator
+}
+
+instance Iterable[Nel] {
```

Comment:
I think all these instances should be in the files where the type and its operations are declared, e.g. Nel.flix

Reply:
Sure, I'll move them